test/spec/exception-handling.wast

```diff
@@ -1,3 +1,153 @@
+(module
+  (tag $e-v)
+  (tag $e-i32 (param i32))
+  (tag $e-f32 (param f32))
+  (tag $e-i32-f32 (param i32 f32))
+
+  (func $throw_single_value (export "throw_single_value")
+    (throw $e-i32 (i32.const 5))
+  )
+
+  (func (export "throw_multiple_values")
+    (throw $e-i32-f32 (i32.const 3) (f32.const 3.5))
+  )
+
+  (func (export "try_table_nothrow") (result i32)
+    (block $tryend (result i32)
+      (drop
+        (block $catch (result i32)
+          (br $tryend
+            (try_table (result i32) (catch $e-i32 $catch)
+              (i32.const 3)
+            )
+          )
+        )
+      )
+      (i32.const 0)
+    )
+  )
+
+  (func (export "try_table_throw_catch") (result i32)
+    (block $tryend (result i32)
+      (drop
+        (block $catch (result i32)
+          (br $tryend
+            (try_table (result i32) (catch $e-i32 $catch)
+              (throw $e-i32 (i32.const 5))
+            )
+          )
+        )
+      )
+      (i32.const 3)
+    )
+  )
+
+  (func (export "try_table_throw_nocatch") (result i32)
+    (block $tryend (result i32)
+      (drop
+        (block $catch (result f32)
+          (br $tryend
+            (try_table (result i32) (catch $e-f32 $catch)
+              (throw $e-i32 (i32.const 5))
+            )
+          )
+        )
+      )
+      (i32.const 3)
+    )
+  )
+
+  (func (export "try_table_throw_catchall") (result i32)
+    (block $tryend (result i32)
+      (block $catch-all
+        (drop
+          (block $catch-f32 (result f32)
+            (br $tryend
+              (try_table (result i32) (catch $e-f32 $catch-f32) (catch_all $catch-all)
+                (throw $e-i32 (i32.const 5))
+              )
+            )
+          )
+        )
+        (br $tryend (i32.const 3))
```

Comment:
```suggestion
        (br $tryend (i32.const 4))
```
It'd be better to make each constant different (the original test has 4 here too)